src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -0,0 +1,720 @@
+                       % (job['sandbox'], job['packageDir'],
+                          'JobPackage.pkl', self.unpacker))
+            argString = "arguments = %s %i\n" \
+                        % (os.path.basename(job['sandbox']), job['id'])
+            jdl.append(argString)
+
+            jdl.extend(self.customizePerJob(job))
+
+            # Transfer the output files
+            jdl.append("transfer_output_files = Report.%i.pkl\n" % (job["retry_count"]))
+
+            # Add priority if necessary
+            task_priority = job.get("taskPriority", self.defaultTaskPriority)
+            try:
+                task_priority = int(task_priority)
+            except:
+                logging.error("Priority for task not castable to an int")
+                logging.error("Not setting priority")
+                logging.debug("Priority: %s", task_priority)
+                task_priority = 0
+
+            prio = 0
+            if job.get('priority'):
+                try:
+                    prio = int(job['priority'])
+                except ValueError:
+                    logging.error("Priority for job %i not castable to an int\n", job['id'])
+                    logging.error("Not setting priority")
+                    logging.debug("Priority: %s", job['priority'])
+                except Exception as ex:
+                    logging.error("Got unhandled exception while setting priority for job %i\n", job['id'])
+                    logging.error(str(ex))
+                    logging.error("Not setting priority")
+
+            jdl.append("priority = %i\n" % (task_priority + prio * self.maxTaskPriority))
+
+            jdl.append("+PostJobPrio1 = -%d\n" % len(job.get('potentialSites', [])))
+            jdl.append("+PostJobPrio2 = -%d\n" % job['taskID'])
+
+            jdl.append("+WMAgent_JobID = %s\n" % job['jobid'])
+
+            jdl.append("job_machine_attrs = GLIDEIN_CMSSite\n")
+
+            ### print all the variables needed for us to rely on condor userlog
+            jdl.append("job_ad_information_attrs = JobStatus,QDate,EnteredCurrentStatus,JobStartDate,DESIRED_Sites,ExtDESIRED_Sites,WMAgent_JobID,MATCH_EXP_JOBGLIDEIN_CMSSite\n")
+
+            jdl.append("Queue 1\n")
+
+        return jdl
+
+    def customizePerJob(self, job):
+        """
+        JDL additions just for this implementation. Over-ridden in sub-classes
+        These are the Glide-in specific bits
+        """
+        jdl = []
+        jobCE = job['location']
+        if not jobCE:
+            # Then we ended up with a site that doesn't exist?
+            logging.error("Job for non-existant site %s", job['location'])
+            return jdl
+
+        if self.submitWMSMode and len(job.get('possibleSites', [])) > 0:
+            strg = ','.join(map(str, job.get('possibleSites')))
+            jdl.append('+DESIRED_Sites = \"%s\"\n' % strg)
+        else:
+            jdl.append('+DESIRED_Sites = \"%s\"\n' % (jobCE))
```

Comment:
Yes. In practice self.submitWMSMode is always True though, since the agents are configured that way. I think this flag was intended for direct condor submissions if someone would want to setup a site where we submit directly to the batch system.

IMO we can get rid of that since in that mode a lot of other code in this plugin makes no sense whatsoever. This is pretty much a gWMS only plugin at this point.

Also not sure if the job submitter even passes jobs for submission that have no possible or potential sites. I think it throws an error on these and immediately fails them for submission. Would allow us to simplify this code here quite a bit.